### PR TITLE
IOS-685: Adding credit cards' validation.

### PR DESCRIPTION
**Pull request by pbruz**
Adding credit cards' validation.

### CODE REVIEW

File: NGRValidator/NGRValidator/Categories/NSString+NGRValidator.m [hunk 1]
```diff
@@ -59,4 +105,43 @@ - (BOOL)evaluateAllowedCharacterSet:(NSCharacterSet *)allowedCharacterSet {
     return (range.location == NSNotFound);
 }
 
+- (BOOL)fulfillsLuhnAlgorithm {
```
Comment (akashivskyy) [inline NGRValidator/NGRValidator/Categories/NSString+NGRValidator.m]: This method being private doesn't make it unsusceptible to collisions. It should be prefixed by `ngr_`, like others.

File: NGRValidator/NGRValidator/Categories/NSString+NGRValidator.m [hunk 2]
```diff
@@ -46,6 +47,51 @@ - (BOOL)ngr_isDecimal {
     return [self evaluateAllowedCharacterSet:[NSCharacterSet decimalDigitCharacterSet]];
 }
 
+
+- (BOOL)ngr_isValidCreditCardNumber {
+    NSString *number = [self stringByReplacingOccurrencesOfString:@" " withString:@""];
```
Comment (akashivskyy) [inline NGRValidator/NGRValidator/Categories/NSString+NGRValidator.m]: I suggest that you also do:

```objc
[... stringByTrimmingCharactersInSet:NSCharacterSet.whitespaceAndNewlineCharacterSet]
```

**Review by akashivskyy — CHANGES_REQUESTED** [review]

File: NGRValidator/NGRValidator/Categories/NSString+NGRValidator.m (hunk 1 above)
Comment (pbruz) [reply NGRValidator/NGRValidator/Categories/NSString+NGRValidator.m]: True that 👍

File: NGRValidator/NGRValidator/Categories/NSString+NGRValidator.m [hunk 3]
```diff
@@ -46,6 +47,52 @@ - (BOOL)ngr_isDecimal {
...
+    }
+
+    NSDate *endOfMonth = [[NSCalendar currentCalendar] dateByAddingUnit:NSCalendarUnitMonth value:1 toDate:expirationDate options:0];
+
+    return [endOfMonth ngr_isLaterThan:[[NSDate alloc] init]];
+}
+
 #pragma mark - Private
 
 - (BOOL)evaluatePattern:(NSString *)pattern {
```
Comment (akashivskyy) [inline NGRValidator/NGRValidator/Categories/NSString+NGRValidator.m]: Can you also add prefixes to the other methods by the way? :)

**Review by akashivskyy — CHANGES_REQUESTED** [review]

**Review by akashivskyy — APPROVED** [review]
🚢 ✨

**Pull request merged**